webapi/src/main/scala/org/knora/webapi/routing/AroundDirectives.scala

```diff
@@ -18,13 +27,38 @@ trait AroundDirectives extends InstrumentationSupport {
   /**
    * When wrapped around a [[akka.http.scaladsl.server.Route]], logs the time it took for the route to run.
    */
-  def logDuration: Directive0 = extractRequestContext.flatMap { ctx =>
+  private val requestCounter: Counter[Long] = Metric.counter("http_request_count")
+  private val requestTimer: Metric[MetricKeyType.Histogram, Duration, MetricState.Histogram] =
+    Metric.timer("http_request_duration", MILLIS, Chunk.iterate(1.0, 5)(_ * 10))
+  def logDuration(implicit runtime: zio.Runtime[Any]): Directive0 = extractRequestContext.flatMap { ctx =>
     val start = System.currentTimeMillis()
     mapResponse { resp =>
-      val took    = System.currentTimeMillis() - start
-      val message = s"[${resp.status.intValue()}] ${ctx.request.method.name} ${ctx.request.uri} took: ${took}ms"
+      val duration = System.currentTimeMillis() - start
+      val message  = s"[${resp.status.intValue()}] ${ctx.request.method.name} ${ctx.request.uri} took: ${duration}ms"
       if (resp.status.isFailure()) metricsLogger.warn(message) else metricsLogger.debug(message)
-      resp
+
+      val path          = replaceIris(ctx.request.uri.path.toString())
+      val httpMethod    = ctx.request.method.name.toUpperCase()
+      val responseCode  = s"${resp.status.intValue()}"
+      val counterMetric = addTags(requestCounter, path, httpMethod, responseCode)
+      val timerMetric   = addTags(requestTimer, path, httpMethod, responseCode)
+      UnsafeZioRun.runOrThrow(
+        (counterMetric.increment *> timerMetric.update(Duration.fromMillis(duration))).ignore.as(resp)
+      )
+    }
+  }
+
+  private def addTags[A, B, C](metric: Metric[A, B, C], path: String, method: String, code: String): Metric[A, B, C] =
+    metric.tagged("method", method).tagged("code", code).tagged("path", path)
+
+  private val iriPattern = Pattern.compile(".*?/(https?:%2F%2F[^/]+)")
+  private def replaceIris(path: String) = {
+    val matcher = iriPattern.matcher(path)
+    if (matcher.find()) {
+      val iri = matcher.group(1)
+      path.replace(iri, "__IRI__")
+    } else {
+      path
     }
   }
```

Comment:
I assume this is done to keep the number of distinct paths as small as possible? In that case it probably needs to be done for the other possible identifiers (shortnames, shortcodes and UUIDs) too.  
I had to do a similar thing in the ZIO HTTP metric.